### PR TITLE
Spring tests

### PR DESCRIPTION
**Pull request by Zel1oy**

### CODE REVIEW

**Review by OlegTuryk — COMMENTED** [review]
Fix conflicts

File: pom.xml [hunk 1]
```diff
@@ -47,37 +63,44 @@
...
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.32</version>
         </dependency>
         <dependency>
-            <groupId>com.mysql</groupId>
-            <artifactId>mysql-connector-j</artifactId>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>test</scope>
```
Comment (Dimagaa) [inline pom.xml:99]: Why do you need h2 database?

File: src/test/resources/db/cart-item-insert.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO cart_items (id, shopping_cart_id, book_id, quantity, is_deleted)
+VALUES (1, 1, 1, 4, false)
```
Comment (Dimagaa) [inline src/test/resources/db/cart-item-insert.sql]: Add an empty line

File: src/test/resources/db/shopping-cart-insert.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO shopping_carts (id, user_id, is_deleted)
+VALUES (1, 2, false)
```
Comment (Dimagaa) [inline src/test/resources/db/shopping-cart-insert.sql]: Add empty line

**Review by Dimagaa — APPROVED** [review]
Well done!

File: src/test/java/com/example/onlinebookstore/controller/CategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(requestDto)))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+        CategoryResponseDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                CategoryResponseDto.class);
+        boolean isEquals = EqualsBuilder.reflectionEquals(
+                expectedCategoryResponseDto, actual, "id");
+        System.out.println(expectedCategoryResponseDto);
```
Comment (andrii-hoienko) [inline src/test/java/com/example/onlinebookstore/controller/CategoryControllerTest.java]: What purpose of this?

File: src/test/java/com/example/onlinebookstore/controller/CategoryControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+    @Sql(scripts = BOOK_CATEGORY_INSERT, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = BOOK_CATEGORY_DELETE, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Test delete with valid request")
+    @Test
+    void deleteCategoryById_validId_deleteCategory() {
+        mockMvc.perform(delete("/categories/2")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNoContent())
+                .andReturn();
+        assertThrows(EntityNotFoundException.class,
```
Comment (andrii-hoienko) [inline src/test/java/com/example/onlinebookstore/controller/CategoryControllerTest.java]: check error message as well

File: src/test/java/com/example/onlinebookstore/service/AuthenticationServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        when(authenticationManager.authenticate(any(
+                UsernamePasswordAuthenticationToken.class))).thenReturn(authentication);
+        when(authentication.getName()).thenReturn(user.getEmail());
+        String expected = "test token";
+        when(jwtUtil.generateToken(anyString())).thenReturn(expected);
+        assertEquals(new UserLoginResponseDto(expected),
+                authenticationService.authenticate(userLoginRequestDto));
+
+        verify(authenticationManager).authenticate(any(UsernamePasswordAuthenticationToken.class));
+        verify(jwtUtil).generateToken(anyString());
```
Comment (andrii-hoienko) [inline src/test/java/com/example/onlinebookstore/service/AuthenticationServiceTest.java]: Not necessary to "verify" mocks that you "when"

File: src/test/java/com/example/onlinebookstore/service/ShoppingCartServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+        when(userService.getAuthenticatedUser()).thenReturn(DEFAULT_USER);
+        when(shoppingCartRepository.findByUserId(anyLong()))
+                .thenReturn(Optional.of(VALID_SHOPPING_CART));
+        when(shoppingCartMapper.toDto(VALID_SHOPPING_CART)).thenReturn(VALID_DTO_RESPONSE);
+
+        ShoppingCartDto actual = shoppingCartService.getShoppingCart();
+        assertNotNull(actual);
+        assertEquals(VALID_DTO_RESPONSE, actual);
+
+        verify(userService, times(1)).getAuthenticatedUser();
```
Comment (andrii-hoienko) [inline src/test/java/com/example/onlinebookstore/service/ShoppingCartServiceTest.java]: times(1) is by default

File: src/test/resources/db/cart-item-insert.sql (hunk 2 above)
Comment (andrii-hoienko) [reply src/test/resources/db/cart-item-insert.sql]: agree

File: src/test/resources/db/shopping-cart-insert.sql (hunk 3 above)
Comment (andrii-hoienko) [reply src/test/resources/db/shopping-cart-insert.sql]: agree

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
GJ! Let`s improve your solution

**Review by nacenik — APPROVED** [review]
LGTM! :)

**Review by nacenik — APPROVED** [review]

**Pull request merged**